src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -134,6 +142,92 @@ public Command stowTurret() {
     return new InstantCommand(() -> setTurretPos(Math.toRadians(0))).withName("StowTurretCommand");
   }
 
+  public Command zeroTurretCommand() {
+    return Commands.runOnce(() -> this.zeroTurret(), this).withName("ZeroTurretCommand");
+  }
+
+  /** Will calculate the current turret position and update encoders and motors off of it. */
+  public void zeroTurret() {
+    double lowGearPosition = lowGearCANcoder.getAbsolutePosition().getValue().doubleValue();
+    double highGearPosition = highGearCANcoder.getAbsolutePosition().getValue().doubleValue();
+    Rotation2d turretRotation = encoderPositionsToTurretRotation(lowGearPosition, highGearPosition);
+
+    lowGearCANcoder.setPosition(
+        turretRotation.getRotations() * Constants.TurretConstants.LOW_GEAR_CAN_CODER_RATIO);
+    highGearCANcoder.setPosition(
+        turretRotation.getRotations() * Constants.TurretConstants.HIGH_GEAR_CAN_CODER_RATIO);
+    turretMotor
+        .getEncoder()
+        .setPosition(
+            turretRotation.getRotations() * Constants.TurretConstants.TURRET_MOTOR_TURRET_RATIO);
+  }
+
+  /**
+   * Calculates the turret zero off of given encoder rotations.
+   *
+   * @param lowGearCANcoderPosition The low gear encoder position, as in the gear with a lower gear
+   *     ratio
+   * @param highGearCANcoderPosition The high gear encoder position, as in the gear with a higher
+   *     gear ratio
+   * @return the calculated turret rotation
+   */
+  public static Rotation2d encoderPositionsToTurretRotation(
+      double lowGearCANcoderPosition, double highGearCANcoderPosition) {
+
+    // This equation is based off of
+    // https://www.geeksforgeeks.org/implementation-of-chinese-remainder-theorem-inverse-modulo-based-implementation/
+    // It is accurate to with in 3.6 deg
+    int gearToothPosition =
+        ((int) (lowGearCANcoderPosition * Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH)
```

Comment:
Don't these need to be rounded instead of truncated?

Reply:
Rounding introduced error near the upper limit. By truncating the issue was resolved with no issue in the lower ranges